chore: Adds some enums to handle types in our models, and adds changes to the models

dtote opened the pull request:
Fix #7 #29 
He añadido enums para los roles de usuario, para el estado de las orders y para las categorias de los productos.

También he modificado los modelos para que las orders se asocien con los usuarios a traves de los ids de los mismos, además de quitar de los users las orders, ya que queremos tenerlas en una tabla separada.

Por último, lo que antes llamábamos cliente ahora lo llamaremos user porque es más cómodo de manejar, ya que los roles de user que manejaremos serán ```client``` y ```restaurant```

dtote committed:
chore: Adds some enums to handle types in our models, and adds changes to the models

jncabdom commented on src/models/user.go:
```diff
@@ -0,0 +1,13 @@
+	HashedPassword string     `json:"hashedPassword"`
+	Phone          int        `json:"phone"`
+	Address        []Address  `json:"address"`
+	Menu           []Product  `json:"menu"`
```
Deberíamos omitir esto para los usuarios con rol cliente no?

```suggestion
	Menu           []Product  `json:"menu"` `json:",omitempty"`
```

dtote replied:
True, muy buena esa 💯

dtote committed:
Update src/models/user.go

Actualizando el menú dentro del modelo de usuario para que si viene vacío se ignore la propiedad menu, ya que eso significaría probablemente que el usuario tiene rol de cliente y no de restaurante

Co-authored-by: Nicolás Cabrera <jncabdom@gmail.com>

dtote committed:
chore: fix minor error
